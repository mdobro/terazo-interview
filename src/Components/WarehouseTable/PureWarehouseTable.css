.warehouseTable {
  overflow: auto;
  height: calc(100vh - 130px);
}

.warehouseHeader th {
  vertical-align: top;
  text-align: center;
}

.pagination {
  display: flex;
  justify-content: center;
  margin-top: 10px;
}
