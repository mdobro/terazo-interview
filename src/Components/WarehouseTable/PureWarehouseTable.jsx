import "./PureWarehouseTable.css";

import React from "react";
import { Table } from "reactstrap";
import WarehouseTableRow from "../WarehouseTableRow/WarehouseTableRow";

const PureWarehouseTable = ({ warehouses, onRowChange, onRowDelete }) => (
  <Table bordered>
    <thead className="warehouseHeader">
      <tr>
        <th rowSpan={2}>Name</th>
        <th rowSpan={2}>Description</th>
        <th colSpan={7}>Address</th>
        <th rowSpan={2}>Actions</th>
      </tr>
      <tr>
        <th>Building Name</th>
        <th>Street Line 1</th>
        <th>Street Line 2</th>
        <th>City</th>
        <th>State/Province</th>
        <th>Postal Code</th>
        <th>Country</th>
      </tr>
    </thead>

    <tbody>
      {warehouses?.map((warehouse) => (
        <WarehouseTableRow
          key={warehouse.warehouseId}
          warehouse={warehouse}
          onRowChange={onRowChange}
          onRowDelete={onRowDelete}
        />
      ))}
    </tbody>
  </Table>
);

PureWarehouseTable.propTypes = {};

PureWarehouseTable.defaultProps = {};

export default PureWarehouseTable;
