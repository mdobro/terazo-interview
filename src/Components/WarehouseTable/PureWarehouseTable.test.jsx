import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import PureWarehouseTable from "./PureWarehouseTable";
import mockData from "../../mockData/warehouses.json";

test("renders", () => {
  const { asFragment } = render(<PureWarehouseTable warehouses={[]} />);
  expect(asFragment()).toMatchSnapshot();
});

test("Search", () => {
  const { getByPlaceholderText, getAllByTestId } = render(
    <PureWarehouseTable warehouses={mockData} />
  );
  const rows = getAllByTestId("warehouseTableRow");
  expect(rows.length === 10);
  const search = getByPlaceholderText("Search");
  userEvent.type(search, "langosh-adams");
  expect(rows.length === 1);
});
