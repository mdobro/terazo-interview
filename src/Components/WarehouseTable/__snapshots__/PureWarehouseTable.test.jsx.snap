// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <table
    class="table table-bordered"
  >
    <thead
      class="warehouseHeader"
    >
      <tr>
        <th
          rowspan="2"
        >
          Name
        </th>
        <th
          rowspan="2"
        >
          Description
        </th>
        <th
          colspan="7"
        >
          Address
        </th>
        <th
          rowspan="2"
        >
          Actions
        </th>
      </tr>
      <tr>
        <th>
          Building Name
        </th>
        <th>
          Street Line 1
        </th>
        <th>
          Street Line 2
        </th>
        <th>
          City
        </th>
        <th>
          State/Province
        </th>
        <th>
          Postal Code
        </th>
        <th>
          Country
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;
